pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/FunctionConfig.java

```diff
@@ -71,18 +72,18 @@
     @NotNull
     @isImplementationOfClasses(implementsClasses = {Function.class, java.util.function.Function.class})
     private String className;
-    @isListEntryCustom(entryValidatorClasses = {ValidatorImpls.TopicNameValidator.class})
-    private Collection<String> inputs;
-    @isMapEntryCustom(keyValidatorClasses = { ValidatorImpls.TopicNameValidator.class },
-            valueValidatorClasses = { ValidatorImpls.SerdeValidator.class }, targetRuntime = ConfigValidation.Runtime.JAVA)
-    @isMapEntryCustom(keyValidatorClasses = { ValidatorImpls.TopicNameValidator.class }, targetRuntime = ConfigValidation.Runtime.PYTHON)
-    private Map<String, String> customSerdeInputs;
-    @isValidTopicName
-    private String topicsPattern;
+
```

Comment:
the whole idea of functionconfig is to be user facing config. For simple things like writing a function taking one input, users should be simply able to say
inputs : topic
as opposed to the new way.
IMO we should support the inputs the way they are.
Inside we could translate inputs into a more comprehensive topicsSchema